processor/k8sattributesprocessor/pod_association.go

```diff
@@ -26,79 +26,91 @@ import (
 	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/k8sattributesprocessor/internal/kube"
 )
 
-// extractPodIds extracts IP and pod UID from attributes or request context.
-// It returns a value pair containing configured label and IP Address and/or Pod UID.
-// If empty value in return it means that attributes does not contains configured label to match resources for Pod.
-func extractPodID(ctx context.Context, attrs pcommon.Map, associations []kube.Association) (string, kube.PodIdentifier) {
+// extractPodIds returns pod identifier for first association matching all sources
+func extractPodID(ctx context.Context, attrs pcommon.Map, associations []kube.Association) kube.PodIdentifier {
 	// If pod association is not set
 	if len(associations) == 0 {
 		return extractPodIDNoAssociations(ctx, attrs)
 	}
 
 	connectionIP := getConnectionIP(ctx)
-	hostname := stringAttributeFromMap(attrs, conventions.AttributeHostName)
 	for _, asso := range associations {
-		// If association configured to take IP address from connection
-		switch {
-		case asso.From == "connection" && connectionIP != "":
-			return k8sIPLabelName, connectionIP
-		case asso.From == "resource_attribute":
-			// If association configured by resource_attribute
-			// In k8s environment, host.name label set to a pod IP address.
-			// If the value doesn't represent an IP address, we skip it.
-			if asso.Name == conventions.AttributeHostName {
-				if net.ParseIP(hostname) != nil {
```

Comment:
Is this this part for parsing host.name of removed intentionally? This seems like unrelated change in the behavior. If it's done intentionally it probably should go through another PR with description why it should be removed

Reply:
I reverted this change. Now it is more clear how it affects the flow.
The hostname is skipped if it is not an ip address. It is not obvious for me, but cleaning it up can be part of another PR, as you said